corehq/apps/locations/middleware.py

```diff
@@ -30,7 +30,7 @@ def process_view(self, request, view_fn, view_args, view_kwargs):
         if not domain or not user or not user.is_member_of(domain):
             # This is probably some non-domain page or a test, let normal auth handle it
             request.can_access_all_locations = True
-        elif user.has_permission(domain, 'access_all_locations'):
+        elif user.has_permission(domain, 'access_all_locations') or user.is_anonymous:
```

Comment:
might be safer to do `or (user.is_anonymous and PUBLISH_CUSTOM_REPORTS.enabled(domain)`